### PR TITLE
Reduce PVC use in CI

### PR DESCRIPTION
**Pull request by Roming22**
PVC are expensive and under quota. This change limits their use to a single PVC per pipeline and only when necessary.

Small refactoring to have a consistent use of the PVC across pipelines

### CODE REVIEW

File: .tekton/tasks/deploy-cluster.yaml [hunk 1]
```diff
@@ -22,7 +21,7 @@ spec:
         - name: CLUSTER_NAME
           value: "$(params.cluster-name)"
         - name: KUBECONFIG_DIR
```
[the remark this answers is not in the document]
Comment (Roming22) [reply .tekton/tasks/deploy-cluster.yaml:23]: Fixed

File: .tekton/tasks/destroy-cluster.yaml [hunk 2]
```diff
@@ -34,9 +33,9 @@ spec:
               name: plnsvc-ci-secret
               key: "PLNSVC_AWS_KEY"
         - name: KUBECONFIG_DIR
```
[the remark this answers is not in the document]
Comment (Roming22) [reply .tekton/tasks/destroy-cluster.yaml:35]: Fixed

**Comment by gabemontero** [comment]
Per the last test failure, looks like you need to update some RBAC @Roming22 

```
+ kubectl -n openshift-config get secret/pull-secret -o yaml
Error from server (Forbidden): secrets "pull-secret" is forbidden: User "system:serviceaccount:tekton-ci:appstudio-pipeline" cannot get resource "secrets" in API group "" in the namespace "openshift-config"
```

No direct refs to `openshift-config` in your PR, so it is not obvious that you added a new ref to that namespace ... maybe the kubeconfig changes altered permissions enough that you lost some existing permissions?

**Comment by xinredhat** [comment]
I think we can merge PR https://github.com/openshift-pipelines/pipeline-service/pull/772 to get rid of this error

**Comment by gabemontero** [comment]
ok @Roming22 with #772 merged go ahead and rebase this PR and let's see what happens with the e2e's

**Comment by xinredhat** [comment]
@Roming22 I think the cause is task `setup-ci-runner` didn't point to the correct kubeconfig file. it should talk to the Hypershift Cluster, but the error is showing that it was trying to talk to the Production cluster.

**Comment by Roming22** [comment]
@xinredhat I found and fixed the issue (I had forgot to replace the workspace name in one of the step env variable).

Please merge if you agree with the changes.

**Review by gabemontero — APPROVED** [review]

**Pull request merged**